src/status.ts

```diff
@@ -0,0 +1,267 @@
+      );
+
+      status.command = {
+        title: "Run bundle install",
+        command: "ruby-lsp.installGems",
+      };
+      return true;
+    }
+    return false;
+  }
+
+  public async addMissingGem(): Promise<boolean> {
+    if (await isGemMissing()) {
+      this.updateStatus(ServerCommand.Error);
+      const status: vscode.LanguageStatusItem = this.createStatusItem(
+        "addMissingGem",
+        "Ruby LSP: Bundle Add",
+        vscode.LanguageStatusSeverity?.Error
+      );
+
+      this.context.subscriptions.push(
+        vscode.commands.registerCommand("ruby-lsp.addMissingGem", () => {
+          status.text = "Ruby LSP: Adding gem...";
+          status.command = undefined;
+          status.busy = true;
+          addGem()
+            .then(() => status.dispose())
+            .catch(() => {});
+        })
+      );
+
+      status.command = {
+        title: "Run bundle add and install",
+        command: "ruby-lsp.addMissingGem",
+      };
+      return true;
+    }
+    return false;
+  }
+
+  private async activateGemOutdatedButton() {
+    const gemOutdated = await isGemOutdated();
+    if (!gemOutdated) {
+      return;
+    }
+
+    const commandId = "updateOutdatedGem";
+
+    const status: vscode.LanguageStatusItem = this.createStatusItem(
+      "updateGem",
+      "Ruby LSP: The gem is not up-to-date",
+      vscode.LanguageStatusSeverity?.Warning
+    );
+
+    this.context.subscriptions.push(
+      vscode.commands.registerCommand(commandId, async () => {
+        status.text = "Ruby LSP: Updating Ruby LSP";
+        status.busy = true;
+        status.command = undefined;
+
+        const result = await updateGem();
+
+        if (result.stderr.length > 0) {
+          status.text = "Ruby LSP: Update failed";
+          status.command = {
+            title: "Try again",
+            command: commandId,
+          };
+
+          if (
+            result.stderr.includes(
+              "Bundler attempted to update ruby-lsp but its version stayed the same"
+            )
+          ) {
+            vscode.window.showWarningMessage(
+              "Could not update the ruby-lsp gem. Is the version in the Gemfile pinned?"
+            );
+          } else {
+            vscode.window.showErrorMessage("Failed to update gem.");
+          }
+        } else {
+          status.dispose();
+        }
+      })
+    );
+
+    status.command = {
+      title: "Update",
+      command: commandId,
+    };
+  }
+
+  private createYjitStatus(ruby: Ruby) {
+    const useYjit = vscode.workspace.getConfiguration("rubyLsp").get("yjit");
+
+    let [major, minor]: number[] = [0, 0];
+
+    if (this.ruby.rubyVersion) {
+      [major, minor] = this.ruby.rubyVersion.split(".").map(Number);
+    }
+
+    const yjit: vscode.LanguageStatusItem =
+      vscode.languages.createLanguageStatusItem("yjit", this.selector);
+    yjit.name = "Ruby LSP Status";
+
+    if (useYjit && ruby.yjitEnabled && [major, minor] >= [3, 2]) {
```

Comment:
Instead of duplicating this check, what do you think about setting some attribute on `Client` like `yjitEnabled`? Then instead of passing the client's context and Ruby to `Status`, we can pass the entire client object and make whatever we need to inspect public.

Reply:
Since that would require a bit of refactoring and this PR is already quite large, would it be best to put this into a separate issue or should it still be addressed here?